README.md

```diff
@@ -1,137 +1,196 @@
+You can then follow these steps:
 
 1. Create a directory in which to build the plugin.
 
-2. Run the CMake GUI or ccmake, specifying your new directory as the build directory and the top
+2. Run the CMake GUI or `ccmake`, specifying your new directory as the build directory and the top
 level directory of this project as the source directory.
 
 3. Press "Configure".  (Do not worry if it produces an error message about not being able to find PyTorch.)
 
-4. Set OPENMM_DIR to point to the directory where OpenMM is installed.  This is needed to locate
+4. Set `OPENMM_DIR` to point to the directory where OpenMM is installed.  This is needed to locate
 the OpenMM header files and libraries.  If you are unsure of what directory this is, the following
 script will print it out.
 
-```
-from simtk.openmm import *
+```python
+from simtk import openmm
 import os
-print(os.path.dirname(version.openmm_library_path))
+print(os.path.dirname(openmm.version.openmm_library_path))
 ```
 
-5. Set PYTORCH_DIR to point to the directory where you installed the LibTorch.
+5. Set `PYTORCH_DIR` to point to the directory where you installed the LibTorch.
 
-6. Set CMAKE_INSTALL_PREFIX to the directory where the plugin should be installed.  Usually,
-this will be the same as OPENMM_DIR, so the plugin will be added to your OpenMM installation.
+6. Set `CMAKE_INSTALL_PREFIX` to the directory where the plugin should be installed.  Usually,
+this will be the same as `OPENMM_DIR`, so the plugin will be added to your OpenMM installation.
 
-7. If you plan to build the OpenCL platform, make sure that OPENCL_INCLUDE_DIR and
-OPENCL_LIBRARY are set correctly, and that NN_BUILD_OPENCL_LIB is selected.
+7. If you plan to build the OpenCL platform, make sure that `OPENCL_INCLUDE_DIR` and
+`OPENCL_LIBRARY` are set correctly, and that `NN_BUILD_OPENCL_LIB` is selected.
 
-8. If you plan to build the CUDA platform, make sure that CUDA_TOOLKIT_ROOT_DIR is set correctly
-and that NN_BUILD_CUDA_LIB is selected.
+8. If you plan to build the CUDA platform, make sure that `CUDA_TOOLKIT_ROOT_DIR` is set correctly
+and that `NN_BUILD_CUDA_LIB` is selected.
 
 9. Press "Configure" again if necessary, then press "Generate".
 
 10. Use the build system you selected to build and install the plugin.  For example, if you
 selected Unix Makefiles, type `make install` to install the plugin, and `make PythonInstall` to
 install the Python wrapper.
 
-Usage
-=====
+Using the OpenMM PyTorch plugin
+===============================
+
+Exporting a PyTorch model for use in OpenMM
+-------------------------------------------
+
+The first step is to create a PyTorch model defining the calculation to perform.  
+It should take particle positions in nanometers (in the form of a `torch.Tensor` of shape `(nparticles,3)` and precision `torch.float64` as its input,
+and return the potential energy in kJ/mol as a `torch.Tensor` of shape `(1,1)` and precision `torch.float64` as output.  
```

Comment:
PyTorch models don't need to specify the precision of their inputs.  They handle whatever is given to them.  In this case, they might be either single or double precision, depending on the mode specified when creating the OpenMM Context.  The precision of the output also isn't important.  It will get cast as necessary.

The output should be a scalar, not a 2D tensor.

Reply:
@peastman : So the entire PyTorch model is in a single common precision? What if you specify arguments with different precisions?

Reply:
@peastman : I've addressed this comment.

Reply:
The model can do whatever it wants with dtypes.  It will receive as input a tensor that could be either single or double precision.  If it just does calculations on it, the output will end up with the same type.

```python
>>> a = torch.tensor([1,2,3], dtype=torch.float32)
>>> torch.sum(a).dtype
torch.float32
```

But if you want to ensure calculations get done in a particular precision, you can include whatever casts you want.

```python
>>> b = a.to(torch.float64)
>>> torch.sum(b).dtype
torch.float64
```

So your model can do whatever it wants.  It just needs to ultimately return the energy, and we don't care what precision the return value has.

Reply:
The docstrings still say, "The forward method returns the (1,1) energy tensor..."

Reply:
Fixed!